app/Model/DecayingModelsFormulas/Base.php

```diff
@@ -0,0 +1,121 @@
+<?php
+abstract class DecayingModelBase
+{
+
+    // Description of the formula. Will be displayed at various places in the web interface
+    // Should be overriden
+    public const DESCRIPTION = '';
+
+    public function checkLoading()
+    {
+        return 'BONFIRE LIT';
+    }
+
+    // Get effective taxonomy ratio based on taxonomies attached to the attribute
+    // Basically, it adapts the ratio defined in the model to fit the actual attached tags 
+    protected function __getRatioScore($model, $tags)
+    {
+        $ratioScore = array();
+        $taxonomy_base_ratio = $model['DecayingModel']['parameters']['base_score_config'];
+        if (empty($taxonomy_base_ratio)) {
+            return array();
+        }
+        $total_score = 0.0;
+        foreach ($tags as $tag) {
+            $namespace_predicate = explode('=', $tag['Tag']['name'])[0];
+            if (isset($taxonomy_base_ratio[$namespace_predicate]) && is_numeric($tag['Tag']['numerical_value'])) {
+                $total_score += floatval($taxonomy_base_ratio[$namespace_predicate]);
+            }
+        }
+        foreach ($tags as $i => $tag) {
+            $namespace_predicate = explode('=', $tag['Tag']['name'])[0];
+            if (isset($taxonomy_base_ratio[$namespace_predicate]) && is_numeric($tag['Tag']['numerical_value'])) {
+                $ratioScore[$namespace_predicate] = floatval($taxonomy_base_ratio[$namespace_predicate]) / $total_score;
+            }
+        }
+        return $ratioScore;
+    }
+
+    // return attribute tag with event tag matching the namespace+predicate overridden
+    protected function __getPrioritizedTag($attribute)
+    {
+        $tags = array();
+        $overridden_tags = array();
+        $temp_mapping = array();
+        if (isset($attribute['EventTag'])) {
+            foreach ($attribute['EventTag'] as $i => $tag) {
+                $tags[] = $tag;
+                $namespace_predicate = explode('=', $tag['Tag']['name'])[0];
+                $temp_mapping[$namespace_predicate] = $i;
+            }
+        }
+        if (isset($attribute['AttributeTag'])) {
+            foreach ($attribute['AttributeTag'] as $tag) {
+                $namespace_predicate = explode('=', $tag['Tag']['name'])[0];
+                if (isset($temp_mapping[$namespace_predicate])) { // need to override event tag
+                    $overridden_tags[] = array(
+                        'EventTag' => $tags[$temp_mapping[$namespace_predicate]],
```

Comment:
UI wise, it should display both overridden `Event` tags and not only the last one